### PR TITLE
rename CMakeTools to CMake Project

### PR DESCRIPTION
**Pull request by elahehrashedi**

### CODE REVIEW

File: src/cmakeProject.ts [hunk 1]
```diff
@@ -531,7 +531,7 @@ export class CMakeTools implements api.CMakeToolsAPI {
      * Dispose the instance
      */
     dispose() {
-        log.debug(localize('disposing.extension', 'Disposing CMakeTools extension'));
+        log.debug(localize('disposing.extension', 'Disposing CMakeProject extension'));
```
Comment (bobbrow) [inline src/cmakeProject.ts:534]: don't change this one

File: src/cmakeProject.ts [hunk 2]
```diff
@@ -881,7 +881,7 @@ export class CMakeTools implements api.CMakeToolsAPI {
      * Second phase of two-phase init. Called by `create`.
      */
     private async init() {
-        log.debug(localize('second.phase.init', 'Starting CMakeTools second-phase init'));
+        log.debug(localize('second.phase.init', 'Starting CMakeProject second-phase init'));
```
Comment (bobbrow) [inline src/cmakeProject.ts:884]: don't rename this one

File: test/util.ts [hunk 3]
```diff
@@ -24,7 +24,7 @@ export async function clearExistingKitConfigurationFile() {
 }
 
 export function getExtension() {
-    const cmt = vscode.extensions.getExtension<CMakeTools>('ms-vscode.cmake-tools');
+    const cmt = vscode.extensions.getExtension<CMakeProject>('ms-vscode.cmake-tools');
```
Comment (bobbrow) [inline test/util.ts:27]: This should probably return the API, but that's currently disabled.  Maybe put a comment here to remind us.

File: test/extension-tests/single-root-UI/test/configure-and-build-kits.test.ts [hunk 4]
```diff
@@ -9,14 +9,14 @@ import {
...
-import CMakeTools from '@cmt/cmakeTools';
+import CMakeProject from '@cmt/cmakeProject';
 
 const workername: string = process.platform;
 
 suite('Build using Kits and Variants', () => {
     let testEnv: DefaultEnvironment;
     let compdb_cp_path: string;
-    let cmakeTools: CMakeTools;
+    let cmakeProjects: CMakeProject;
```
Comment (bobbrow) [inline test/extension-tests/single-root-UI/test/configure-and-build-kits.test.ts:19]: `cmakeProject` (without an 's')

File: test/extension-tests/successful-build/test/configure-and-build.test.ts [hunk 5]
```diff
@@ -15,7 +15,7 @@ import * as path from 'path';
 const workername: string = process.platform;
 
 suite('Build', () => {
-    let cmt: CMakeTools;
+    let cmt: CMakeProject;
```
Comment (bobbrow) [inline test/extension-tests/successful-build/test/configure-and-build.test.ts:18]: rename `cmt`?

File: test/extension-tests/successful-build/test/debugger.test.ts [hunk 6]
```diff
@@ -1,19 +1,19 @@
...
-import { CMakeTools } from '@cmt/cmakeTools';
+import { CMakeProject } from '@cmt/cmakeProject';
 import { DefaultEnvironment, expect, getFirstSystemKit } from '@test/util';
 //import sinon = require('sinon');
 import * as fs from 'fs';
 import * as path from 'path';
 
 suite('Debug/Launch interface', () => {
-    let cmt: CMakeTools;
+    let cmt: CMakeProject;
```
Comment (bobbrow) [inline test/extension-tests/successful-build/test/debugger.test.ts:9]: rename `cmt`?

File: test/extension-tests/successful-build/test/cmake.test.ts [hunk 7]
```diff
@@ -1,8 +1,8 @@
-import { CMakeTools } from '@cmt/cmakeTools';
+import { CMakeProject } from '@cmt/cmakeProject';
 import { clearExistingKitConfigurationFile, DefaultEnvironment, expect, getFirstSystemKit } from '@test/util';
 
 suite('cmake', () => {
-    let cmt: CMakeTools;
+    let cmt: CMakeProject;
```
Comment (bobbrow) [inline test/extension-tests/successful-build/test/cmake.test.ts:5]: rename `cmt`?

**Review by bobbrow — APPROVED** [review]
Please also rename `cmt` in the tests

File: test/util.ts (hunk 3 above)
Comment (elahehrashedi) [reply test/util.ts:27]: Copied som parts of the other util.thisExtension function for this function.

**Review by bobbrow — APPROVED** [review]

**Pull request merged**